sensors/Pms.py

```diff
@@ -0,0 +1,234 @@
+        self.__device = device
+        self.__timeout = timeout
+        self.__baud_rate = baud_rate
+        self.__reading = ()
+
+
+    # Sets the GPIO to be in input mode
+    def connect_to_sensor(self) -> serial:
+        f"""
+        Connects the software to the sensor hardware.
+        
+        :return: {serial} 
+        """
+        try:
+            ser = serial.Serial(self.__device, self.__baud_rate, timeout=self.__timeout, parity=serial.PARITY_NONE)
+        except SerialException:
+            Sensor.print_formatted_data({"error": "connecting to sensor"})
+            exit(1)
+        else:
+            return ser
+
+    def get_pm_1_raw(self)->float:
+        return self.__reading[DATA_INDICES['PM1.0']]
+    
+    def get_pm_2_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['PM2.5']]
+    
+    def get_pm_10_raw(self)->float:
+        return self.__reading[DATA_INDICES['PM2.5']]
+    
+    def get_pm_1(self)->float:
+        return μgm3_to_gpl(self.__reading[DATA_INDICES['PM1.0']])
+    
+    def get_pm_2_5(self)->float:
+        return μgm3_to_gpl(self.__reading[DATA_INDICES['PM2.5']])
+    
+    def get_pm_10(self)->float:
+        return μgm3_to_gpl(self.__reading[DATA_INDICES['PM2.5']])
+    
+    
+    
+    #greater than 
+    def get_gt_0_3_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt0.3μm']]
+    
+    def get_gt_0_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt0.5μm']]
+    
+    def get_gt_1_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt1.0μm']]
+    
+    def get_gt_2_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt2.5μm']]
+
+    def get_gt_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt5.0μm']]
+    
+    def get_gt_10_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt10μm']]
+    
+    # processed
+    def get_gt_0_3(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt0.3μm']])
+    
+    def get_gt_0_5(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt0.5μm']])
+    
+    def get_gt_1(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt1.0μm']])
+    
+    def get_gt_2_5(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt2.5μm']])
+
+    def get_gt_5(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt5.0μm']])
+    
+    def get_gt_10(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt10μm']])
+    
+    
+
+	# Get raw data
+    def get_raw_data(self)-> float:
+        
+        ser = self.connect_to_sensor()
+        
+        start = time.time()
+
+        sof_index = 0
+
+        while True:
+            elapsed = time.time() - start
+            if elapsed > 5:
+                raise ReadTimeoutError("PMS5003 Read Timeout: Could not find start of frame")
+
+            sof = ser.read(1)
```

Comment:
I think `sof` should be renamed to provide greater semantic meaning

Reply:
Changed to "start_of_file"